inc/ticket.class.php

```diff
@@ -1630,6 +1652,21 @@ function prepareInputForAdd($input) {
          $input['itilcategories_id_code'] = ITILCategory::getById($cat_id)->fields['code'];
       }
 
+      // Set default contract if not specified
+      if (!isset($input['_contracts_id'])) {
+         $input['_contracts_id'] = Entity::getDefaultContract($this->input['entities_id'] ?? $this->fields['entities_id'] ?? false);
```

Comment:
1. `$this->fields` will never be set in `prepareInputForAdd`.
2. If `$input['entities_id']` is not defined, ticket will be created in root entity, so you could fallback to `0`.
```suggestion
         $input['_contracts_id'] = Entity::getDefaultContract($input['entities_id'] ?? 0);
```

Reply:
Done in a5da574 .